ci: samples: frdm_k64f build failure

**Describe the bug**
Latest main doesn't build sample/subsys/mgmt/mcumgr/smp_svr leading to failed CI on PRs.
It looks like it is related to NET_IF in the smp_udp.c file as reported by the compiler errors.

It might affect other boards I cannot tell.

**To Reproduce**
```
west build -p -b frdm_k64f samples/subsys/mgmt/mcumgr/smp_svr -T sample.mcumgr.smp_svr.udp
-- west build: making build dir /home/dev/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/dev/zephyrproject/zephyr/samples/subsys/mgmt/mcumgr/smp_svr
-- CMake version: 3.22.1
-- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/dev/.cache/zephyr
-- Zephyr version: 3.6.99 (/home/dev/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: frdm_k64f, qualifiers: mk64f12
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/home/dev/zephyr-sdk-0.16.5)
-- Found toolchain: zephyr 0.16.5 (/home/dev/zephyr-sdk-0.16.5)
-- Found Dtc: /home/dev/zephyr-sdk-0.16.5/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/dev/zephyrproject/zephyr/boards/nxp/frdm_k64f/frdm_k64f.dts
-- Generated zephyr.dts: /home/dev/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/dev/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/dev/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/dev/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/dev/zephyrproject/zephyr/boards/nxp/frdm_k64f/frdm_k64f_defconfig'
Merged configuration '/home/dev/zephyrproject/zephyr/samples/subsys/mgmt/mcumgr/smp_svr/prj.conf'
Merged configuration '/home/dev/zephyrproject/zephyr/samples/subsys/mgmt/mcumgr/smp_svr/overlay-udp.conf'
Configuration saved to '/home/dev/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/dev/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /home/dev/zephyr-sdk-0.16.5/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/dev/zephyr-sdk-0.16.5/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
Load components for MK64F12:
driver_common component is included.
driver_reset component is included.
device_CMSIS component is included.
CMSIS_Include_core_cm component is included.
device_system component is included.
driver_rnga component is included.
driver_enet component is included.
driver_uart component is included.
driver_flash component is included.
driver_port component is included.
driver_sysmpu component is included.
-- Using ccache: /usr/bin/ccache
-- Including signing script: /home/dev/zephyrproject/zephyr/cmake/mcuboot.cmake
CMake Warning at /home/dev/zephyrproject/zephyr/cmake/mcuboot.cmake:28 (message):
  Neither CONFIG_MCUBOOT_GENERATE_UNSIGNED_IMAGE or
  CONFIG_MCUBOOT_SIGNATURE_KEY_FILE are set, the generated build will not be
  bootable by MCUboot unless it is signed manually/externally.
Call Stack (most recent call first):
  /home/dev/zephyrproject/zephyr/cmake/mcuboot.cmake:189 (zephyr_mcuboot_tasks)
  /home/dev/zephyrproject/zephyr/CMakeLists.txt:1904 (include)


-- Configuring done
-- Generating done
-- Build files have been written to: /home/dev/zephyrproject/zephyr/build
-- west build: building application
[1/230] Preparing syscall dependency handling

[3/230] Generating include/generated/version.h
-- Zephyr version: 3.6.99 (/home/dev/zephyrproject/zephyr), build: v3.6.0-4302-gde526632ae7a
[223/230] Building C object zephyr/subsys/mgmt/mcumgr/transport/CMakeFiles/subsys__mgmt__mcumgr__transport.dir/src/smp_udp.c.obj
FAILED: zephyr/subsys/mgmt/mcumgr/transport/CMakeFiles/subsys__mgmt__mcumgr__transport.dir/src/smp_udp.c.obj 
ccache /home/dev/zephyr-sdk-0.16.5/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCPU_MK64FN1M0VLL12 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/dev/zephyrproject/zephyr/include -I/home/dev/zephyrproject/zephyr/build/zephyr/include/generated -I/home/dev/zephyrproject/zephyr/soc/nxp/kinetis -I/home/dev/zephyrproject/zephyr/soc/nxp/kinetis/k6x/. -I/home/dev/zephyrproject/zephyr/soc/nxp/kinetis/common -I/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/util/include -I/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/include -I/home/dev/zephyrproject/zephyr/subsys/net/l2 -I/home/dev/zephyrproject/zephyr/subsys/net/conn_mgr/. -I/home/dev/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/dev/zephyrproject/zephyr/modules/cmsis/. -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MK64F12 -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MK64F12/drivers -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/rnga -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/enet -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/uart -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/flash -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/port -I/home/dev/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/sysmpu -I/home/dev/zephyrproject/zephyr/modules/hal_nxp/. -I/home/dev/zephyrproject/modules/lib/zcbor/include -isystem /home/dev/zephyrproject/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /home/dev/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/home/dev/zephyr-sdk-0.16.5/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/dev/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/dev/zephyrproject/zephyr/samples/subsys/mgmt/mcumgr/smp_svr=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/dev/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/dev/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/subsys/mgmt/mcumgr/transport/CMakeFiles/subsys__mgmt__mcumgr__transport.dir/src/smp_udp.c.obj -MF zephyr/subsys/mgmt/mcumgr/transport/CMakeFiles/subsys__mgmt__mcumgr__transport.dir/src/smp_udp.c.obj.d -o zephyr/subsys/mgmt/mcumgr/transport/CMakeFiles/subsys__mgmt__mcumgr__transport.dir/src/smp_udp.c.obj -c /home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c: In function 'smp_udp_open_iface':
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:275:13: warning: implicit declaration of function 'net_if_is_up' [-Wimplicit-function-declaration]
  275 |         if (net_if_is_up(iface)) {
      |             ^~~~~~~~~~~~
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:277:21: warning: implicit declaration of function 'net_if_flag_is_set'; did you mean 'net_buf_frag_insert'? [-Wimplicit-function-declaration]
  277 |                 if (net_if_flag_is_set(iface, NET_IF_IPV4) &&
      |                     ^~~~~~~~~~~~~~~~~~
      |                     net_buf_frag_insert
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:277:47: error: 'NET_IF_IPV4' undeclared (first use in this function)
  277 |                 if (net_if_flag_is_set(iface, NET_IF_IPV4) &&
      |                                               ^~~~~~~~~~~
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:277:47: note: each undeclared identifier is reported only once for each function it appears in
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:284:47: error: 'NET_IF_IPV6' undeclared (first use in this function)
  284 |                 if (net_if_flag_is_set(iface, NET_IF_IPV6) &&
      |                                               ^~~~~~~~~~~
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c: In function 'smp_udp_open':
/home/dev/zephyrproject/zephyr/subsys/mgmt/mcumgr/transport/src/smp_udp.c:339:17: warning: implicit declaration of function 'net_if_foreach' [-Wimplicit-function-declaration]
  339 |                 net_if_foreach(smp_udp_open_iface, NULL);
      |                 ^~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/dev/zephyrproject/zephyr/build
```

**Expected behavior**
The build is successful and CI passes.

**Impact**
No more CI passing.


**Environment (please complete the following information):**
 - OS: Xubuntu 22.04
 - Toolchain (e.g Zephyr SDK, ...)
 - Zephyr: Commit de526632ae7a58eb323e68b7f44c76bd6a79c627

--- a/subsys/mgmt/mcumgr/transport/src/smp_udp.c
+++ b/subsys/mgmt/mcumgr/transport/src/smp_udp.c
@@ -14,6 +14,7 @@
 #include <zephyr/kernel.h>
 #include <zephyr/init.h>
 #include <zephyr/net/socket.h>
+#include <zephyr/net/net_if.h>
 #include <zephyr/mgmt/mcumgr/mgmt/mgmt.h>
 #include <zephyr/mgmt/mcumgr/smp/smp.h>
 #include <zephyr/mgmt/mcumgr/transport/smp.h>

